dangerzone/logic.py

```diff
@@ -36,3 +38,35 @@ def __init__(self) -> None:
 
         # Load settings
         self.settings = Settings(self)
+
+        self.documents: List[Document] = []
+
+    def add_document(
+        self, input_filename: str, output_filename: Optional[str] = None
+    ) -> None:
+        doc = Document(input_filename)
+        if output_filename:
+            doc.output_filename = output_filename
+        else:
+            doc.set_default_output_filename()
```

Comment:
I think it's simpler to just:

1. Pass the `Document` instances here. No need for this class to perform the initialization.
2. Change the `Document` class to always have a default for output filenames (`the `self.set_default_output_filename()`).

This will help in the case we add more arguments to `Document.__init__`, or if we need subclasses of `Document` at some point.